qiskit_experiments/quantum_volume/qv_experiment.py

```diff
@@ -0,0 +1,249 @@
+            )
+        # count the number of previous trials.
+        # assuming that all the data in experiment data is QV data.
+        self._previous_trials = len(experiment_data.data())
+
+        # ideal circuits are used for getting the ideal state using the simulator, without the
+        # changes that the transpiler might add to the circuit in order to improve it's performance
+        circuits = self.circuits(backend)
+        ideal_circuits = []
+        for circ in circuits:
+            circ.metadata["is_simulation"] = False
+            # return new circuit without the measurements
+            ideal_circuits.append(circ.remove_final_measurements(inplace=False))
+            ideal_circuits[-1].metadata["is_simulation"] = True
+        # transpile circuits
+        circuits = transpile(circuits, backend, **self.transpile_options.__dict__)
+
+        # Run circuits on backend
+        run_opts = copy.copy(self.run_options)
+        run_opts.update_options(**run_options)
+        run_opts = run_opts.__dict__
+
+        if isinstance(backend, LegacyBackend):
+            qobj = assemble(circuits, backend=backend, **run_opts)
+            job = backend.run(qobj)
+        else:
+            job = backend.run(circuits, **run_opts)
+
+        # Add Jobs to ExperimentData
+        experiment_data.add_data(job)
+
+        sim_data = self._get_ideal_data(ideal_circuits, **run_opts)
+        # Add Jobs to the simulation data
+        simulation_data.add_data(sim_data)
+
+        # Queue analysis of data for when job is finished
+        if analysis and self.__analysis_class__ is not None:
+            self.run_analysis(experiment_data, simulation_data=simulation_data)
+
+        self._simulation_data = simulation_data
+        # Return the ExperimentData future
+        return experiment_data
+
+    def add_trials(self, additional_trials):
+        """
+        Add more trials to the experiment
+        Args:
+            additional_trials (int): The amount of trials to add
+        """
+        new_trials = self.experiment_options.trials + additional_trials
+        self.set_experiment_options(trials=new_trials)
+
+    @property
+    def trials(self):
+        """Return number of trials in the experiment"""
+        return self.experiment_options.trials
+
+    @property
+    def simulation_data(self):
+        """Return the ideal data of the experiment"""
+        return self._simulation_data
+
+    def _get_ideal_data(self, circuits, **run_options):
+        """
+        in case the user does not have aer installed - use Terra to calculate the ideal state
+        Args:
+            circuits: the circuits to extract the ideal data from
+        Returns:
+            dict: data object with the circuit's metadata
+                  and the probability for each state in each circuit
+        """
+        if self._simulation_backend:
+            for circuit in circuits:
+                circuit.save_probabilities()
```

Comment:
I am not sure, but it's worth to check if it will be better to use `save_probabilites_dict()` and `state_vector.probabilities_dict()` methods (in case the QV is high)